tests/common.go

```diff
@@ -2892,6 +2894,193 @@ func DeleteBucket(provider string, bucketName string) {
 	})
 }
 
+//HaIncreaseRebootTargetNode repl increase and reboot target node
+func HaIncreaseRebootTargetNode(event *EventRecord, ctx *scheduler.Context, v *volume.Volume, storageNodeMap map[string]node.Node) {
+
+	Step(
+		fmt.Sprintf("repl increase volume driver %s on app %s's volume: %v and reboot target node",
+			Inst().V.String(), ctx.App.Key, v),
+		func() {
+
+			replicaSets, err := Inst().V.GetReplicaSets(v)
+			if err == nil {
+				replicaNodes := replicaSets[0].Nodes
+				var newReplID string
+				var newReplNode node.Node
+
+				//selecting the target node for repl increase
+				for nID, node := range storageNodeMap {
+					nExist := false
+					for _, id := range replicaNodes {
+						if nID == id {
+							nExist = true
+							break
+						}
+					}
+					if !nExist {
+						newReplID = nID
+						newReplNode = node
+						break
+					}
+				}
+
+				currRep, err := Inst().V.GetReplicationFactor(v)
+
+				if err != nil {
+					err = fmt.Errorf("error getting replication factor for volume %s, Error: %v", v.Name, err)
+					logrus.Error(err)
+					UpdateOutcome(event, err)
+					return
+				}
+
+				//if repl is 3 cannot increase repl for the volume
+				if currRep == 0 || currRep == 3 {
+					err = fmt.Errorf("cannot perform repl incease as current repl factor is %d", currRep)
+					logrus.Error(err)
+					UpdateOutcome(event, err)
+					return
+				}
+
+				Step(
+					fmt.Sprintf("repl increase volume driver %s on app %s's volume: %v",
+						Inst().V.String(), ctx.App.Key, v),
+					func() {
+						logrus.Infof("Increasing repl with target node  [%v]", newReplID)
+						err = Inst().V.SetReplicationFactor(v, currRep+1, []string{newReplID}, false)
+						if err != nil {
+							logrus.Errorf("There is an error increasing repl [%v]", err.Error())
+							UpdateOutcome(event, err)
+						}
+					})
+
+				if err == nil {
+					Step(
+						fmt.Sprintf("reboot target node %s while repl increase is in-progres",
+							newReplNode.Hostname),
+						func() {
+
+							logrus.Info("Waiting for 10 seconds for re-sync to initialize before target node reboot")
```

Comment:
Instead of waiting 10 seconds, can we add a code to make sure re-sync to initialize before rebooting the node?

Reply:
checking for re-sync status will be little tricky in some cases where volume size is small  re-sync will happen very fast.
 And we are checking for volume is clean or not after ha-increase and reboot to verfiy succesfull ha-increase